crowbar_framework/app/models/service_object.rb

```diff
@@ -822,8 +822,19 @@ def apply_role(role, inst, in_queue)
     # We also check that all nodes we'll require are in the ready state.
     #
 
-    # Initialize variables used in ensure at the end of the method
-    chef_daemon_nodes = []
+    nodes_without_admin = NodeObject.find_all_nodes.reject do |node|
```

Comment:
Since we're also excluding windows nodes, would it make sense to rename this variable to something like `nodes_to_lock` or something closer to what it really is?